Xquery mediator with registry reference throws exception when invoked

**Description:**
In micro-integrator, when an XQuery mediator refers to an XQuery resource in the registry, it throws the below exception. This occurs as the file extension map does not contain the .xqy file extension with XQuery content type.

```
[2019-08-05 17:23:08,229] [micro-integrator] ERROR - XQueryMediator Unable to execute the query 
java.lang.RuntimeException: ContentID is null
	at org.apache.axiom.om.impl.llom.OMTextImpl.getDataHandler(OMTextImpl.java:404)
	at org.apache.synapse.mediators.xquery.XQueryMediator.performQuery(XQueryMediator.java:242)
	at org.apache.synapse.mediators.xquery.XQueryMediator.mediate(XQueryMediator.java:127)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71)
	at org.apache.synapse.mediators.filters.FilterMediator.mediate(FilterMediator.java:169)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
	at org.apache.synapse.core.axis2.ProxyServiceMessageReceiver.receive(ProxyServiceMessageReceiver.java:224)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.axis2.transport.local.LocalTransportReceiver.processMessage(LocalTransportReceiver.java:170)
	at org.apache.axis2.transport.local.LocalTransportReceiver.processMessage(LocalTransportReceiver.java:82)
	at org.wso2.carbon.core.transports.local.CarbonLocalTransportSender.finalizeSendWithToAddress(CarbonLocalTransportSender.java:45)
	at org.apache.axis2.transport.local.LocalTransportSender.invoke(LocalTransportSender.java:77)
	at org.apache.axis2.engine.AxisEngine.send(AxisEngine.java:442)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.send(DynamicAxisOperation.java:185)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.executeImpl(DynamicAxisOperation.java:167)
	at org.apache.axis2.client.OperationClient.execute(OperationClient.java:149)
	at org.apache.synapse.core.axis2.Axis2FlexibleMEPClient.send(Axis2FlexibleMEPClient.java:634)
	at org.apache.synapse.core.axis2.Axis2Sender.sendOn(Axis2Sender.java:85)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.send(Axis2SynapseEnvironment.java:569)
	at org.apache.synapse.endpoints.AbstractEndpoint.send(AbstractEndpoint.java:398)
	at org.apache.synapse.endpoints.AddressEndpoint.send(AddressEndpoint.java:65)
	at org.apache.synapse.mediators.builtin.SendMediator.mediate(SendMediator.java:123)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
	at org.apache.synapse.core.axis2.ProxyServiceMessageReceiver.receive(ProxyServiceMessageReceiver.java:224)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.synapse.transport.passthru.ServerWorker.processEntityEnclosingRequest(ServerWorker.java:415)
	at org.apache.synapse.transport.passthru.ServerWorker.run(ServerWorker.java:152)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

The following response is observed.

```
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
   <soapenv:Body>
      <soapenv:Fault>
         <faultcode xmlns:soap11Env="http://schemas.xmlsoap.org/soap/envelope/">soap11Env:Server</faultcode>
         <faultstring>Unable to execute the query</faultstring>
         <detail>
            <ns:GetShoppingBagV2Fault xmlns:ns="ikea.com/system/iip/GetShoppingBagV2/1.0/">
               <TransactionError>
                  <SourceId>VPCService</SourceId>
                  <ErrorCode>0</ErrorCode>
                  <ErrorMessage>org.apache.synapse.SynapseException: Unable to execute the query 
	at org.apache.synapse.mediators.xquery.XQueryMediator.handleException(XQueryMediator.java:633)
	at org.apache.synapse.mediators.xquery.XQueryMediator.mediate(XQueryMediator.java:134)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71)
	at org.apache.synapse.mediators.filters.FilterMediator.mediate(FilterMediator.java:169)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
	at org.apache.synapse.core.axis2.ProxyServiceMessageReceiver.receive(ProxyServiceMessageReceiver.java:224)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.axis2.transport.local.LocalTransportReceiver.processMessage(LocalTransportReceiver.java:170)
	at org.apache.axis2.transport.local.LocalTransportReceiver.processMessage(LocalTransportReceiver.java:82)
	at org.wso2.carbon.core.transports.local.CarbonLocalTransportSender.finalizeSendWithToAddress(CarbonLocalTransportSender.java:45)
	at org.apache.axis2.transport.local.LocalTransportSender.invoke(LocalTransportSender.java:77)
	at org.apache.axis2.engine.AxisEngine.send(AxisEngine.java:442)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.send(DynamicAxisOperation.java:185)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.executeImpl(DynamicAxisOperation.java:167)
	at org.apache.axis2.client.OperationClient.execute(OperationClient.java:149)
	at org.apache.synapse.core.axis2.Axis2FlexibleMEPClient.send(Axis2FlexibleMEPClient.java:634)
	at org.apache.synapse.core.axis2.Axis2Sender.sendOn(Axis2Sender.java:85)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.send(Axis2SynapseEnvironment.java:569)
	at org.apache.synapse.endpoints.AbstractEndpoint.send(AbstractEndpoint.java:398)
	at org.apache.synapse.endpoints.AddressEndpoint.send(AddressEndpoint.java:65)
	at org.apache.synapse.mediators.builtin.SendMediator.mediate(SendMediator.java:123)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
	at org.apache.synapse.core.axis2.ProxyServiceMessageReceiver.receive(ProxyServiceMessageReceiver.java:224)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.synapse.transport.passthru.ServerWorker.processEntityEnclosingRequest(ServerWorker.java:415)
	at org.apache.synapse.transport.passthru.ServerWorker.run(ServerWorker.java:152)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: ContentID is null
	at org.apache.axiom.om.impl.llom.OMTextImpl.getDataHandler(OMTextImpl.java:404)
	at org.apache.synapse.mediators.xquery.XQueryMediator.performQuery(XQueryMediator.java:242)
	at org.apache.synapse.mediators.xquery.XQueryMediator.mediate(XQueryMediator.java:127)
	... 33 more</ErrorMessage>
                  <ErrorDesc>Unable to execute the query</ErrorDesc>
               </TransactionError>
            </ns:GetShoppingBagV2Fault>
         </detail>
      </soapenv:Fault>
   </soapenv:Body>
</soapenv:Envelope>
```

**Related Issues:**
https://github.com/wso2/micro-integrator/issues/175

--- a/migration/registry/from-mi-1.0.0-to-later/src/main/java/RegistryMigrator.java
+++ b/migration/registry/from-mi-1.0.0-to-later/src/main/java/RegistryMigrator.java
@@ -140,6 +140,7 @@ public class RegistryMigrator {
         map.put(".xslt", "application/xslt+xml");
         map.put(".zip", "application/zip");
         map.put(".wsdl", "WSDL");
+        map.put(".xqy", "XQuery");
 
         return map;
     }

